### DISCUSSION TITLE
Feature request (config): Title/subtitle template

### DISCUSSION THREAD

**Discussion by FantomeBeignet** [ghostty-org/ghostty#4096] [Feature Requests, Ideas]
I think it could be nice to give users more freedom to customize the title and subtitle. Setting the `title` (and possibly `subtitle`) config options with some kind of string interpolation syntax and predefined variables would allow for that.

Here is what I have in mind: 

- Give the user access to some predefined variables such as `command`, `program` or `cwd` (for the current running command, program and current working directory). For example, running the command `ping 1.1.1.1` in the home directory would set `command="ping 1.1.1.1"`, `program=ping`, `cwd=~`.
- Allow the `title` (and possibly `subtitle`) config variables to be set with some kind of string interpolation syntax such as `title = "Ghostty: ${cwd}"`.

This feature could also make for a better search experience using the tab overview (for example, having vim open from two different directories in two separate tabs currently doesn't allow the search to differentiate between the two tabs).

Slightly related to #3570 and #4095.

**Comment by pluiedev** [comment]
I like this proposal but I have reservations about introducing a string interpolation syntax. Would be a bit too much work on our end since we have to parse interpolations and manage to string them together — Zig's string interpolation happens entirely at compile time, so we cannot use its formatting mechanisms. Also the subtitle is only relevant on GTK (macOS windows can't have one).

Perhaps a better solution would be to allow for users to customize our shell integration into displaying any format that they want — then we don't have to interpret the format ourselves and only the shell does.

**Reply by pluiedev** [reply]
That is literally not what this discussion is about. You're hijacking a feature request to suggest something that's completely unrelated. Maybe instead of using a completely clueless LLM you could have some basic decency and actually read the conversation within this thread?

**Comment by ansel1** [comment]
I generally agree that dynamism in the tab title should be handled inside the shell, so I'm not sure a full templating language in ghostty is needed.

The **one** exception I keep running up against is that I really want to add a fixed prefix to window and tab titles in ghostty (not inside the shell) to make it easier to remember what project/task I'm doing in a window or in a tab.  Like, the only template variable I really want is like `%s = whatever the shell wants the title to be`, so that I can set my window title to:

    Project Nexus - %s
    
Or set a tab title to:

    Fixing #4225 - %s
    
As mentioned in other discussion (e.g. https://github.com/ghostty-org/ghostty/discussions/10350), with AI agents, I tend to have many windows, tabs, and tab splits open and active all the time.  I end up treating windows and tabs like workspaces: a tab workspace might have many splits open, but the entire tab is dedicated to a single project or task.  

I still want my shell to have control some of the title content, like showing the current process and cwd of the currently focussed split, but I also want to add a fixed label to the tab which is outside the context of the shell.

Maybe it's just a separate ghostty field, like "Window Label" and "Tab Label", which is rendered adjacent to the Window/Tab Title?
